app/models/concerns/has_custom_fields.rb

```diff
@@ -142,6 +142,11 @@ def reload(options = nil)
     super
   end
 
+  def on_custom_fields_change
```

Comment:
Custom fields unfortunately do not rely on ActiveRecord callbacks since we write directly to the database.